kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -392,7 +390,7 @@ impl<'tcx> GotocCtx<'tcx> {
             let cases = targets
                 .iter()
                 .map(|(c, bb)| {
-                    Expr::int_constant(c, self.codegen_ty(switch_ty))
+                    Expr::int_constant(c, switch_ty.clone())
```

Comment:
nit: I don't think you need the clone here.

Reply:
I also thought it shouldn't be needed, but the compiler corrected me :)
```
error[E0507]: cannot move out of `switch_ty`, a captured variable in an `FnMut` closure
   --> kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs:393:43
    |
372 |         let switch_ty = v.typ().clone();
    |             --------- captured outer variable
...
392 |                 .map(|(c, bb)| {
    |                      --------- captured by this `FnMut` closure
393 |                     Expr::int_constant(c, switch_ty)
    |                                           ^^^^^^^^^ move occurs because `switch_ty` has type `cprover_bindings::goto_program::Type`, which does not implement the `Copy` trait
```
It's because it's used in a closure which might be called multiple times.